### PR TITLE
feat: added a script for importing useres and a user delete endpoint

### PR DESCRIPTION
**Pull request by markusahlstrand**
- Changed the updateUser handler so it also can delete users.
- Added a delete endpoint for users
- Added a script for importing users from a auth0 json dump.

### CODE REVIEW

File: scripts/upload-users.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+const fs = require("fs");
+const readline = require("readline");
+
+const token = "Add token here...";
```
Comment (danielgent) [inline scripts/upload-users.js:4]: :smile: 
image

File: scripts/upload-users.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        authorization: `Bearer ${token}`,
+        "content-type": "application/json",
+      },
+      body,
+    },
+  );
+
+  if (!response.ok) {
+    console.log(
+      `Status: ${response.status}, with error ${await response.text()}`,
```
Comment (danielgent) [inline scripts/upload-users.js:34]: I didn't know you could `await` inside a template literal like this :thinking:

File: scripts/upload-users.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    tags: [],
+    givenName: user.given_name,
+    familyName: user.family_name,
+    id: user.user_id,
+    createdAt: user.created_at,
+    modifiedAt: user.updated_at,
+  });
+
+  const response = await fetch(
+    "https://auth2.sesamy.dev/tenants/AH_1eG1-Ouam8jRlSd1fI/users",
```
Comment (danielgent) [inline scripts/upload-users.js:21]: Is this right?  I imagined we'd be putting each user into the relevant tenant as they're no longer all in the same tenant :thinking: 

If it is, what about extracting out `AH_1eG1-Ouam8jRlSd1fI` as an env var above so we don't accidentally push users to the wrong place (at some point you'll run this on prod I imagine?)

File: wrangler.toml [hunk 4]
```diff
@@ -17,6 +17,7 @@ route = { pattern = "auth2.sesamy.dev/*", zone_id = "b56a76d859fe2ed6a8d77270cd2
 [vars]
 IMAGE_PROXY_URL = "https://imgproxy.prod.sesamy.cloud"
 ISSUER = "https://auth2.sesamy.dev/"
+# ISSUER = "http://localhost:8787/"
```
Comment (danielgent) [inline wrangler.toml:20]: :eyes:

Comment (danielgent) [reply wrangler.toml:20]:
```suggestion
```

File: scripts/upload-users.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    input: fileStream,
+    crlfDelay: Infinity, // so it can handle even `\r\n` ending lines correctly
+  });
+
+  let counter = 0;
+
+  for await (const line of rl) {
+    // if (counter++ > 3) {
+    //   throw new Error("test");
+    // }
```
Comment (danielgent) [inline scripts/upload-users.js:52]:
```suggestion
```
:question:

Comment (danielgent) [reply scripts/upload-users.js:52]: TIL that comments can be linked to multiple lines by clicking and dragging :exclamation:

File: src/routes/tsoa/users.ts [hunk 6]
```diff
@@ -153,17 +154,46 @@ export class UsersController extends Controller {
...
+      throw new NotFoundError();
+    }
+
+    // Fetch the user from durable object
+    const user = env.userFactory.getInstanceByName(
+      getId(tenantId, dbUser.email),
+    );
 
-    return result;
+    return user.delete.mutate();
```
Comment (danielgent) [inline src/routes/tsoa/users.ts:197]: I'm a bit confused as to the flow here and the duplication of the SQL :thinking: 

This route calls `user.delete.mutate` which then calls  `sendUserEvent(UserEvent.userDeleted)` which then calls `deleteUser` which runs the SQL to delete again :thinking:

File: scripts/upload-users.js (hunk 3 above)
Comment (markusahlstrand) [reply scripts/upload-users.js:21]: Yepp, sounds good. This is very quick and dirty

File: src/routes/tsoa/users.ts (hunk 6 above)
Comment (markusahlstrand) [reply src/routes/tsoa/users.ts:197]: Yepp, so the Durable Objects has the master-data. So we can sync from DO's to sql but not the other way around. Currently the DO's have no way of looking up a user id, so we use the sql for that.. Maybe we should add support for that in the DO's but it's not something we need to do when logging in so not super critical.

**Pull request merged**